File: benchmarks/tsb/bench_str_isalnum_isnumeric.ts
/**
 * Benchmark: str_isalnum_isnumeric — str.isalnum and str.isnumeric on 100k strings
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => (i % 3 === 0 ? `abc123` : i % 3 === 1 ? `12345` : `hello!`));
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.str.isalnum();
  s.str.isnumeric();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.str.isalnum();
  s.str.isnumeric();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "str_isalnum_isnumeric",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_str_isalnum_isnumeric.py
"""Benchmark: str_isalnum_isnumeric — str.isalnum and str.isnumeric on 100k strings"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = ["abc123" if i % 3 == 0 else ("12345" if i % 3 == 1 else "hello!") for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.str.isalnum()
    s.str.isnumeric()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.str.isalnum()
    s.str.isnumeric()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "str_isalnum_isnumeric",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
